src/server/journal/streamer.cc

```diff
@@ -208,6 +208,8 @@ void RestoreStreamer::Run() {
     if (fiber_cancelled_)
       return;
     cursor = pt->TraverseBuckets(cursor, [&](PrimeTable::bucket_iterator it) {
+      std::lock_guard guard(big_value_mu_);
```

Comment:
I believe the big_value_mu_ should protect only the WriteBucket to make sure that we do not write to the serializer from different fibers once we yield

Reply:
Idealy it would be inside the WriteBucket function but I think we might have some failures in the flow of CVCUponInsert which could be fixed but not in this PR for sure

Reply:
you will also need to call db_slice_->BlockingCounter() , check out SliceSnapshot::BucketSaveCb

Reply:
This last comment makes me think that we are missing cluster migration test for dragonfly running in cache mode and for testing expire logic. Lets open a seprate task for that and do this in a separate PR

Reply:
I mimicked the locking pattern in `snapshot.cc`.

There, we acquire the lock before calling `FlushChangeToEarlierCallbacks()`, which I imagine is needed? If we'll lock inside `WriteBucket()` the call to `FlushChangeToEarlierCallbacks()` will be unguarded.

Also, by locking inside `WriteBucket()` we risk releasing the lock when we call it multiple times in `CVCUponInsert()` (so that the operation will no longer be atomic, if another fiber locks the mutex in between)

I added "locking" the blocking counter, good catch!

Will file an issue shortly.